controllers/cluster_controller.go

```diff
@@ -250,13 +250,19 @@ func (r *ClusterReconciler) upgradeControlPlane(ctx context.Context, cluster *ca
 	// Create new MachineTemplate and update KubeadmControlPlane to use it, if updating k8s.
 	// Update KubeadmControlPlane label with CACP release number.
 	expectedK8sVersion := fmt.Sprintf("v%s", getComponentVersion(giantswarmRelease, "kubernetes"))
+	expectedETCDVersion := fmt.Sprintf("v%s", getComponentVersion(giantswarmRelease, "etcd"))
 	expectedOSVersion := getComponentVersion(giantswarmRelease, "image")
 	expectedMachineImage := MachineImagesByK8sVersion[expectedK8sVersion][expectedOSVersion]
 	currentMachineImage, _, err := unstructured.NestedString(infraMachineTemplate.Object, strings.Split(KindToMachineImagePath[kcp.Spec.InfrastructureTemplate.Kind], ".")...)
 	if err != nil {
 		return false, errors.Wrapf(err, "failed to read current machine image from infrastructure machine template %q", infraMachineTemplate.GetName())
 	}
 
+	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.Local.ImageTag != expectedETCDVersion {
```

Comment:
Are we sure the etcd version on the Release CR is equivalent to the imageTag? (not sure if we enforce it)

Reply:
Actually, it's not equivalent, but close. Instead of relying on upstream docker images, I believe we want to use our own images created by `retagger` upload to `quay.io/giantswarm`.
For some reason I don't know, not all `etcd` versions have public images. On top of that, the images created by `retagger` don't follow exactly the same version scheme than upstream images.

So for example, this exists `k8s.gcr.io/etcd:3.4.2`, but this doesn't `k8s.gcr.io/etcd:3.4.14`. Not only that, this exists `quay.io/giantswarm/etcd:v3.4.2` but this doesn't `quay.io/giantswarm/etcd:3.4.2`.

That's why in the code I'm adding a leading `v` to the version found in the `Release`.